caddyhttp/httpserver/proxyprotocol.go

```diff
@@ -0,0 +1,53 @@
+package httpserver
+
+import (
+	"fmt"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/armon/go-proxyproto"
+)
+
+type filteredPROXYListener struct {
+	net.Listener
+	timeout time.Duration
+	subnets []*net.IPNet
+}
+
+func newFilteredPROXYListener(ln net.Listener, headerTimeout time.Duration, subnets string) (*filteredPROXYListener, error) {
```

Comment:
I know this is unexported (which is good), but I think this should still be godoc-commented for clarity. What does this method do?

Also document the filteredPROXYListener type above.